### PR TITLE
Expose the CURLOPT_HTTP09_ALLOWED libcurl option

### PR DESCRIPTION
**Pull request by jbangelo**
This PR addresses my feature request in #436. It adds the ability to set the [`CURLOPT_HTTP09_ALLOWED`](https://curl.se/libcurl/c/CURLOPT_HTTP09_ALLOWED.html) option. Unfortunately there are still use cases for this option.

I took my best guess as to how to properly implement it. Please let me know if there's anything that needs to be changed, or even if exposing this option is acceptable.

### CODE REVIEW

File: src/easy/handle.rs [hunk 1]
```diff
@@ -1280,6 +1280,11 @@ impl Easy {
         self.inner.pipewait(wait)
     }
 
+    /// Same as [`Easy2::http_09`](struct.Easy2.html#method.http_09)
+    pub fn http_09(&mut self, allow: bool) -> Result<(), Error> {
```
Comment (sagebind) [inline src/easy/handle.rs:1284]: Nit: I might name this `http_09_allowed` which would more closely match the libcurl option name.

**Comment by sagebind** [comment]
Thanks! The implementation looks correct to me, just some minor suggestions.

**Comment by jbangelo** [comment]
@sagebind I've updated the method names as you suggested. Thanks for the quick feedback!

**Review by sagebind — APPROVED** [review]
Looks great!

**Pull request merged**